### DISCUSSION TITLE
Any example/tutorial on online beam-search decoding?

### DISCUSSION THREAD

**Discussion by xizi642** [flashlight/flashlight#484] [Q&A] [answered]
Hi, I saw the document of flashlight claims that the decoders are now supporting online decoding.  It consumes small chunks of emissions of audio as input. As far as I can understand, it requires the test binary build/bin/fl_asr_test to generate the emissions of audio, and then decode binary build/bin/fl_asr_decode to do beam-search decoding. And "At the time we want to have a look at the transcript so far, we may get the best transcript and prune the hypothesis space and keep decoding further". How can I do that?  couldn't find any flag to achieve this. 
Is there a way for flashlight to consume small chunks of audio data on the fly, generate emissions of this audio in a queue and the use the beam-search decode binary to get the best transcript so far.

**Comment by vineelpratap** [comment, marked as answer]
Hi,
   For doing online inference we created a library called wav2letter@anywhere. 
   Checkout the resources 
   1. Codebase - https://github.com/facebookresearch/wav2letter/tree/master/recipes/streaming_convnets/inference 
   2. Build Fixes - https://github.com/facebookresearch/wav2letter/pull/930/
   3. Wiki - https://github.com/facebookresearch/wav2letter/wiki/Inference-Framework
